apis/v1/secretproviderclass_types.go

```diff
@@ -46,13 +46,32 @@ type SecretObject struct {
 	Data        []*SecretObjectData `json:"data,omitempty"`
 }
 
+type Secret struct {
+	// name of the object to sync
+	ObjectName string `json:"objectName,omitempty"`
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
+	JsonPath string `json:"jsonPath,omitempty"`
+}
+
+type TransformOptions struct {
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
```

Comment:
I think we can actually omit this just determine behavior based on whether `JsonPath` is populated.